docs/source/differentiation/differentiation.rst

```diff
@@ -0,0 +1,49 @@
+
+Gradient, Jacobian and Hessian
+==============================
+
+Estimagic wraps `numdifftools <https://pypi.org/project/numdifftools/>`_ to provide functions to calculate very precise gradients, jacobians and hessians of functions. The precision is achieved by evaluating numerical derivatives at different step sizes and using richardson extrapolations. While this increases the computational cost, it works for any function, whereas other approaches that would yield a similar precision, like complex step derivatives have stronger requirements.
+
+
+.. automodule:: estimagic.differentiation.differentiation
+    :members:
```

Comment:
the Returns part of this docstring is not very informative ("the index is the index of params_sr")

Reply:
fixed.